src/core/framegraphes/forwardrenderer.cpp

```diff
@@ -856,14 +900,20 @@ QVector<AbstractRenderStage *> ForwardRenderer::renderStages() const
 QSize ForwardRenderer::currentSurfaceSize() const
 {
     QSize size;
-    auto surface = m_surfaceSelector->surface();
+    QObject *surface = m_surfaceSelector->surface();
+    Qt3DRender::QRenderTarget *renderTarget = m_renderTargetSelector->target();
 
-    if (auto window = qobject_cast<QWindow *>(surface))
-        size = window->size();
-    else if (qobject_cast<QOffscreenSurface *>(surface))
-        size = m_surfaceSelector->externalRenderTargetSize();
-    else
-        qCWarning(kuesa) << Q_FUNC_INFO << "Unexpected surface type for surface " << surface;
+    if (renderTarget) {
+        Qt3DRender::QAbstractTexture *texture = findRenderTargetTexture(renderTarget, Qt3DRender::QRenderTargetOutput::Color0);
```

Comment:
What happens if the render target doesn't have the Color0 attachment?

Reply:
Changed the patch so that in that case it falls back to rendering to a surface (if any is given)